pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -330,46 +336,49 @@ private void updateSegmentMetrics(String tableNameWithType, TableConfig tableCon
         }
       }
 
-      int numEVReplicas = 0;
+      int numEVReplicasUp = 0;
       if (externalView != null) {
         Map<String, String> stateMap = externalView.getStateMap(segment);
         if (stateMap != null) {
           for (Map.Entry<String, String> entry : stateMap.entrySet()) {
-            String state = entry.getValue();
-            if (state.equals(SegmentStateModel.ONLINE) || state.equals(SegmentStateModel.CONSUMING)) {
-              numEVReplicas++;
+            String serverInstanceId = entry.getKey();
+            String segmentState = entry.getValue();
+            if (isServerQueryable(serverQueryInfoFetcher.getServerQueryInfo(serverInstanceId))
+              && (segmentState.equals(SegmentStateModel.ONLINE) || segmentState.equals(SegmentStateModel.CONSUMING))) {
```

Comment:
(totally optional)  We can probably make this a set called acceptableStates and do acceptableStates.contains(segmentState)